cluster-autoscaler/proposals/buffers.md

```diff
@@ -0,0 +1,574 @@
+given type or pods that use too many resources. 
+
+Since buffers are not pods they will not be accounted by the quota system and so
+any user who is able to create buffers will be able to create a buffer of any
+size.
+
+For the initial implementation on buffers the users should use other tools to
+limit the size of the cluster (max size on the GKE node pool/CCC or karpenter
+pool, max total size). Once we launch we will reassess the need for other
+mechanisms for buffers. 
+
+Two options that should be considered are:
+- [recommended at the moment, to reevaluate before implementation] integrating
+with k8s resource quotas mechanism: make the buffer controller quota aware, 
+apply the quota when translating the buffer config to a pod spec and 
+periodically check the total resources used by the namespace
+- create a separate API surface (like `BufferPolicy`) to manage what buffers can
+be created. This may be needed if we would like to offer different quotas
+depending on the `Buffer` type.
+
+Other considered alternatives: 
+[Buffers and k8s quotas](https://docs.google.com/document/d/1dNxpAb5_fSv4SOIOUYsBYKpE-o_qsCUiSUTS5wCXiT8)
+
+## Dependencies
+
+##### Does this feature depend on any specific services running in the cluster?
+
+Yes, compatible node autoscaler and buffer controller (the latter run as part of
+cluster autoscaler in the reference implementation)
+
+## Scalability
+
+##### Will enabling / using this feature result in any new API calls?
+
+Yes, buffer controller will list cluster objects. No for external calls (outside
+of the cluster).
+
+##### Will enabling / using this feature result in introducing new API types?
+
+Yes (NodeAutoscalingBuffer)
+
+##### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+##### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No
+
+##### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+##### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+##### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+The same way as scheduling a higher number of pods.
+
+# Alternatives
+
+## Do nothing
+
+Since there is already well know path of creating balloon pods/deployments we
+could decide to do nothing and point users to these “mechanisms”
+
+### Why not?
+
+Managing balloon pods proves to be problematic - you need to keep them in sync
+with the workload they need to scale for or in sync with the VM shape (to avoid
+creating too many of these).
+
+We gathered feedback from the customers that maintenance of balloon pods and
+sizing them is something that should be simplified. Additionally if we have a
+buffer object we can add additional feature that are requested for example:
```

Comment:
I think this is also doable with `CronJob` balloon pods, so not really an argument against them.